kpi/management/commands/provision_server.py

```diff
@@ -0,0 +1,184 @@
+        )
+
+    def handle(self, *args, **kwargs):
+        command = kwargs.get('command')
+
+        if command == 'socialapp':
+            self.handle_socialapp(kwargs)
+        elif command == 'config':
+            self.handle_config(kwargs)
+        else:
+            self.stdout.write(self.style.ERROR('No valid subcommand provided'))
+
+    def handle_socialapp(self, kwargs):
+        provider = kwargs['provider']
+        provider_id = kwargs['provider_id']
+        name = kwargs['name']
+        client_id = kwargs['client_id']
+        secret = os.getenv('SOCIAL_APP_SECRET') or kwargs['secret']
+        key = kwargs['key']
+        settings_json = kwargs['settings']
+
+        try:
+            settings = json.loads(settings_json)
+        except TypeError:
+            raise json.JSONDecodeError
+
+        social_app_data = {
+            'provider': provider,
+            'provider_id': provider_id,
+            'name': name,
+            'client_id': client_id,
+            'secret': secret,
+            'key': key,
+            'settings': settings,
+        }
+
+        social_app, created = SocialApp.objects.get_or_create(
+            defaults=social_app_data
+        )
+
+        if not created:
+            self.stdout.write(
+                f'Social app for {social_app.name} already exists'
+            )
+        else:
+            self.stdout.write(
+                f'Successfully created social app for {social_app.name}'
+            )
+
+        social_app_custom_data_exists = SocialAppCustomData.objects.filter(
+            social_app=social_app
+        ).exists()
+
+        if not social_app_custom_data_exists:
+            SocialAppCustomData.objects.create(
+                social_app=social_app,
+                is_public=False,
+            )
+            self.stdout.write(
+                f'Successfully created custom data for {social_app.name}'
+            )
+        else:
+            self.stdout.write(
+                f'Custom data for {social_app.name} already exists'
+            )
+
+    def handle_config(self, kwargs):
+        config_kv_pairs = kwargs['config_kv_pairs']
+        for pair in config_kv_pairs:
+            key, value = pair.split('=')
+            if hasattr(config, key):
+                try:
+                    if value.lower() == 'true':
+                        value = True
+                    elif value.lower() == 'false':
+                        value = False
+                    else:
+                        # Specific handling for fields taking JSON arrays
+                        if key in [
+                            'PROJECT_METADATA_FIELDS',
+                            'USER_METADATA_FIELDS',
+                        ]:
+                            if value.startswith('[') and value.endswith(']'):
+                                try:
+                                    value = json.loads(value)
+                                    value = json.dumps(value)
```

Comment:
Interesting way to validate json, I imagine it works fine. I don't think it's possible to error in the dumps line, so that can be moved outside of the try, except.

Or actually I think you could achieve the validation with 

```
try:
   json.loads(value)
except json.JSONDecodeError ...
```
you don't really need the value, right? You're just converting to and then from json to validate it I assume.

If you end up keeping this code - I might try to break it out so it's not so nested. I don't typically say one must "Don't repeat yourself" when repeating just twice. But in this case it might help reduce the nesting.